tests/yast2_ncurse/yast2_dns_server.pm

```diff
@@ -0,0 +1,101 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: yast2-dns-server bind
+# Summary: Ensure that all combinations of running/stopped and active/inactive can be set for yast2 dns-server
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+use base "y2_module_consoletest";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use Utils::Architectures;
+use y2_module_basetest 'is_network_manager_default';
+use constant RETRIES => 5;
+use serial_terminal 'select_serial_terminal';
+
+sub verify_named_service {
+    my ($round_no, $combo_value1, $combo_value2, $ifactive, $ifenable) = @_;
+
+    YaST::Module::open(module => 'dns-server', ui => 'ncurses');
+    if (is_network_manager_default) {
+        $testapi::distri->get_dns_server()->get_Network_Manager_Warning_page();
+        save_screenshot;
+        $testapi::distri->get_dns_server()->accept_continue();
+        save_screenshot;
+    }
+    $testapi::distri->get_dns_server()->get_Main_page();
+    save_screenshot;
+    $testapi::distri->get_dns_server()->set_service_action_main($combo_value1);
+    save_screenshot;
+    $testapi::distri->get_dns_server()->set_service_autostart_main($combo_value2);
+    save_screenshot;
+    if ($round_no == 5) {
+        $testapi::distri->get_dns_server()->press_abort();
+        save_screenshot;
+        $testapi::distri->get_dns_server()->accept_yes();
+    } else {
+        $testapi::distri->get_dns_server()->accept_apply();
+        wait_still_screen 10;
+        save_screenshot;
+        $testapi::distri->get_dns_server()->accept_ok();
+    }
+
+    select_console 'root-console';
+    systemctl 'is-active named', expect_false => $ifactive;
+    systemctl 'is-enabled named', expect_false => $ifenable;
+}
+
+sub run {
+    my $self = shift;
+
+    select_console 'root-console';
+    zypper_call 'in yast2-dns-server bind ' . $self->firewall, timeout => 180;
+    # Pretend this is the 1st execution (could not be the case if yast2_cmdline was executed before)
+    assert_script_run 'rm -f /var/lib/YaST2/dns_server';
+
+    record_info '1st run', '[wizard-like interface] -> service active & enabled';
```

Comment:
we don't need to use 'record_info' but split this big test module into small module, each one follow the AAA pattern, Arrange, Act and Assert, in our case Arrange for opening the module, Act for configuring all the option and press apply and Assert for validating with command what was configured.